### ISSUE TITLE
monitor.clickOverlay is undefined

### ISSUE THREAD

**Issue by Lythenas** [paperwm/PaperWM#627]
**Describe the bug**
I haven't noticed an immediate problem that this causes, but I noticed some errors in the log.

```
gnome-shell[6654]: meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
gnome-shell[6654]: meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
gnome-shell[6654]: JS ERROR: TypeError: monitor.clickOverlay is undefined
                   switchWorkspace@/home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/tiling.js:2069:13
                   init/<@/home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/tiling.js:1699:47
                   activateWithFocus@/home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/tiling.js:278:28
                   select@/home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/stackoverlay.js:195:20
                   ClickOverlay/<@/home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/stackoverlay.js:125:18
gnome-shell[6654]: Object .Gjs_ui_messageTray_Notification (0x5624b2bc5800), has been already disposed — impossible to emit any signal on it. This might be caused by the object having been destroyed from C code using something such >
gnome-shell[6654]: == Stack trace for context 0x5624b228d4b0 ==
gnome-shell[6654]: #0   5624b2692fd8 i   resource:///org/gnome/shell/ui/messageTray.js:493 (3dab52f51f10 @ 69)
gnome-shell[6654]: #1   5624b2692f30 i   resource:///org/gnome/shell/ui/messageTray.js:489 (3dab52f51ec0 @ 56)
gnome-shell[6654]: #2   5624b2692ea8 i   resource:///org/gnome/shell/ui/calendar.js:796 (3dab52f5e100 @ 22)
gnome-shell[6654]: == Stack trace for context 0x5624b228d4b0 ==
gnome-shell[6654]: #0   5624b2692fd8 i   resource:///org/gnome/shell/ui/messageTray.js:494 (3dab52f51f10 @ 84)
gnome-shell[6654]: #1   5624b2692f30 i   resource:///org/gnome/shell/ui/messageTray.js:489 (3dab52f51ec0 @ 56)
gnome-shell[6654]: #2   5624b2692ea8 i   resource:///org/gnome/shell/ui/calendar.js:796 (3dab52f5e100 @ 22)
```

Also (not sure if that is related) sometimes after lock and unlock my second monitor stops being managed by PaperWM. I haven't noticed that on Gnome 44. Only on my Gnome 42 machine. There are also no errors in the log when that happens. So not sure how to debug this...

**To Reproduce**
1. Unclear, I just saw the errors in the logs.

**Expected behavior**
Errors should not happen.

**Screenshots**

**System information:**
Please execute `./gather-system-info.sh` in you PaperWM clone and paste the output below.

```
Distribution: Ubuntu
GNOME Shell 42.9
Display server: Wayland
PaperWM branch/tag: gnome-44
PaperWM commit: 8c465f69aadb377d047e5bb95b906706eaf51e70
Enabled extensions:
- paperwm@paperwm.github.com
- search-light@icedman.github.com
- tophat@fflewddur.github.io
- arch-update@RaphaelRochet
- date-menu-formatter@marcinjakubowski.github.com
- expandable-notifications@kaan.g.inam.org
- lockkeys@vaina.lt
- ubuntu-appindicators@ubuntu.com
```

**Additional context**

**Comment by jtaala** [comment]
```
gnome-shell[6654]: meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
gnome-shell[6654]: meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
```

That's interesting, why can't gnome find your monitor?  Could during connection of a monitor (which hasn't been registered / pickedup yet?).

I can put a check on `monitor.Clickoverlay` (which is prob a good idea anyways) but that would be addressing the exception but maybe not the root cause.

Let me know if you can find any way to reproduce those errors.  Would likely be coming out of `disable` (e.g. lock/unlock) or when connecting a monitor (or could be something else entirely?).

**Comment by jtaala** [comment]
```
gnome-shell[6654]: Object .Gjs_ui_messageTray_Notification (0x5624b2bc5800), has been already disposed — impossible to emit any signal on it. This might be caused by the object having been destroyed from C code using something such >
```
above makes it look like log-out cycle? (e.g. gnome logging-out / shutting things down... monitor not available, paperwm disabling things... monitor not available... hence clickoverlay not availble) - so a check on there would prob make sense.

**Comment by Lythenas** [comment]
Unfortunately I wasn't able to reproduce the error but I did see this issue again

> Also (not sure if that is related) sometimes after lock and unlock my second monitor stops being managed by PaperWM. I haven't noticed that on Gnome 44. Only on my Gnome 42 machine. There are also no errors in the log when that happens. So not sure how to debug this...

I rolled back a couple of commits and after that I did not see the error anymore. I will investigate this again next week and try to reproduce the clickoverlay error.

**Comment by jtaala** [comment]
Thanks @Lythenas.

If you can get any semi-reproducible steps for the monitor losing control, we can then bisect and find out which commit might have caused this.

Let me know the commit you've rolled back to and I'll look at diffs to see if anything could be the culprit there.

**Comment by jtaala** [comment]
Closed by #633
